src/net.cpp

```diff
@@ -1841,19 +1841,22 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         if (interruptNet)
             return;
 
-        // Add seed nodes if DNS seeds are all down (an infrastructure attack?).
-        // Note that we only do this if we started with an empty peers.dat,
-        // (in which case we will query DNS seeds immediately) *and* the DNS
-        // seeds have not returned any results.
-        if (addrman.size() == 0 && (GetTime() - nStart > 60)) {
-            static bool done = false;
-            if (!done) {
+        // If we started with an empty peers.dat, we will query DNS seeds immediately.
+        // Here we allow 60 seconds for the DNS seeds to return results. If they do not
+        // return any results, or if they were disabled with -dnsseed=0, we add
+        // the fixed seeds.
+        bool dnsseed = gArgs.GetBoolArg("-dnsseed", DEFAULT_DNSSEED);
+        if (addrman.size() == 0 &&
+            ((GetTime() - nStart > 60) || !dnsseed)) {
```

Comment:
Might be outside the scope of this PR, but ISTM`GetTime()` is deprecated; see `src/util/time.h`.